src/activityMonitor.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-console */
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+
+class ActivityMonitor {
+	private readonly refreshRate: number;
+	private readonly logInterval: number;
+	private readonly logFilePath: string;
+	private readonly systemCommand: string;
+
+	constructor(refreshRate: number, logInterval: number, logFilePath: string) {
+		const windowCommand = `powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + ' ' + $_.CPU + ' ' + $_.WorkingSet }"`;
+		const LinuxAndMacOSCommand = 'ps -A -o %cpu,%mem,comm | sort -nr | head -n 1';
+
+		this.systemCommand = process.platform === 'win32' ? windowCommand : LinuxAndMacOSCommand;
+		this.refreshRate = refreshRate;
+		this.logInterval = logInterval;
+		this.logFilePath = logFilePath;
+	}
+
+	private writeToLog(output: string): void {
+		const unixTime = Math.floor(Date.now() / 1000);
+		try {
+			if (!fs.existsSync(this.logFilePath)) {
+				fs.writeFileSync(this.logFilePath, '');
+			}
+			fs.appendFileSync(this.logFilePath, `${unixTime} : ${output}\n`);
+		} catch (error) {
+			console.error('Error writing to log file:', error);
+		}
+	}
+
+	private logSystemActivity(): void {
+		try {
+			const output = child_process.execSync(this.systemCommand).toString().trim();
+			this.writeToLog(output);
+		} catch (error) {
+			console.error('Error logging system activity:', error);
+		}
+	}
+
+	private displaySystemActivity(): void {
+		try {
+			const output = child_process.execSync(this.systemCommand).toString().trim();
+			process.stdout.write(output + '\r');
```

Comment:
This can cause some problems. For example try to run the code below:
```typescript
setInterval(() => {
  const length = Math.floor(Math.random() * 100); //Generate random length from 0 to 100
  let text = '';

  const characters =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';

  for (let i = 0; i < length; i++) {
    text += characters.charAt(Math.floor(Math.random() * characters.length)); // Generate random string
  }
  process.stdout.write(text + '\r');
}, 3000);

```
`stdout.write(text, "\r")` writes the text to console and the moves the cursor to beginning of the current line, but does not clear the line. So if the second output length is less then the previous line the remaining part from previous line still will be visible. So I would suggest you to use:
```typescript
     process.stdout.clearLine();  // Clear the whole line
     process.stdout.cursorTo(0);  // Move the cursor to the start of the line
     process.stdout.write(output);  // Write the new output
```